test/integration/fake_upstream.cc

```diff
@@ -647,6 +706,18 @@ void FakeUpstream::onRecvDatagram(Network::UdpRecvData& data) {
   received_datagrams_.emplace_back(std::move(data));
 }
 
+AssertionResult FakeUpstream::runOnDispatcherThreadAndWait(std::function<AssertionResult()> cb) {
```

Comment:
Should this have a timeout to avoid test hard timeout? It could be hard coded to 15s if that is easier?

Reply:
Added a timeout which results in a RELEASE_ASSERT if hit.  The callbacks run on the dispatcher should be really fast, so it taking an extended period of time indicates that something has gone really wrong.  Cancelling the work is not an option since it is hard to tell how far the callback got before the timeout happens.

Returning AssertionResult is not a good option because there are FakeUpstream functions that wait for an HTTP upstream connection with short timeout in a loop and interpret the returned AssertionResult as a retryable failure.